### PR TITLE
Create Virtual Network PS Script using template.json

### LINKED ISSUE #71: [Task 3] Use New-AzResourceGroupDeployment to create a VNet
Use New-AzRespourceGroupDeployment with template.json and without parameters.json

| Date | Activity| Time |
| :---:        |     :---:      |     :---: |
| 1/24   | Research AZ Resource Deployment    | 1 hr    |
| 1/26   | Created task on Github    | 5 min    |
| 1/27   | Started updating existing script, researched correct template, troubleshooting    | 3 hrs    |
| 1/28   | Updates to script, more research     | 2 hrs    |
| 1/29   | Updated script with more dynamic variables, troubleshooting, testing PR for Juno    | 4hrs    
| 1/29   | Finished script, troubleshooting     | 2.5hrs    |

### PR DESCRIPTION
**Pull request by sunshinejen**
Task #71 
This script creates a Virtual Network with one subnet

1. Open Powershell
2. Make sure you navigate to where script is located in the automation/vnet directory and enter ./deploy-vnet.ps1
-Enter information that is prompted, you will need an existing Azure subscription and resource group name
**You will be asked to supply values for the following parameters:**
- TentantID
- Username and Password
If you enter a resource group that does not exist 
**You will be asked to supply values for the following parameters:**
-Location: Location where new Resource group
will be deployed  

If you enter a resource group that exists
**You will be asked to supply values for the following parameters:**

- ResourceGroupName: Name of your existing resource group
- VirtualNetworkName: Name of your vnet
- SubNetworkName: Name of your subnet

On Azure confirm Virtual Network has been created with your given parameters
If you are testing make sure to delete any created resources

### CODE REVIEW

File: automation/vnet/deploy-vnet.ps1 [hunk 1]
```diff
@@ -4,55 +4,70 @@
...
   This script  requires an existing Azure subscription and Resource group.
-.PARAMETER RGName
-    Required. Name of existing Resource  
-.PARAMETER Location
-    Required. Name of location of Resource eg: westus2     
-.PARAMETER VNetName
-    Required. Name of desired Virtual Network Name Eg: [project]-[resource-type]-[environment]-[location]-[other-stuff]
-.PARAMETER VNetPrefix
-    Required. Desired address space for Virtual Network eg: eg: 10.0.0.0/16
+.PARAMETER TenantId
```
Comment (OyunominJuno) [inline automation/vnet/deploy-vnet.ps1:7]: in the documentation of the PR, this parameter was not mentioned.

File: automation/vnet/deploy-vnet.ps1 [hunk 2]
```diff
@@ -4,55 +4,70 @@
...
-  Creation Date:  01/15/21
-  Purpose: VNet creation Script
+  Creation Date:  01/27/21
+  Purpose: Virtual Network Automation Script
 #>
 
 [cmdletbinding()]
 Param(
-    [Parameter(Mandatory=$true)]
+    [Parameter(Mandatory=$true,HelpMessage='Enter your TenantId')]
```
Comment (OyunominJuno) [inline automation/vnet/deploy-vnet.ps1]: it is good to know about the HelpMessage. will use that in the future

File: automation/vnet/deploy-vnet.ps1 [hunk 3]
```diff
@@ -4,55 +4,70 @@
...
+$Credential = Get-Credential
+Connect-AzAccount -Credential $Credential -Tenant $TenantId -ServicePrincipal
+
+# Get existing resource group
+$Location= Get-AzResourceGroup -Name $ResourceGroupName -ErrorVariable notPresent -ErrorAction SilentlyContinue
+
+#if errorVariable is notPresent VNet will not be created
+if ($notPresent)
+{
+    Write-Output "Existing resource group not found"
```
Comment (OyunominJuno) [inline automation/vnet/deploy-vnet.ps1]: if the resource group is not found, you can create one with the name entered and ask for a location. You can refer to my code for that.

File: automation/vnet/deploy-vnet.ps1 [hunk 4]
```diff
@@ -4,55 +4,70 @@
...
+}
+else
+{
+
+#location selected resource group
+$RGLocation= $Location.location
+
+#file path for -TemplateFile
+$FilePath = "./template.json"
```
Comment (OyunominJuno) [inline automation/vnet/deploy-vnet.ps1]: what if the user wants to use different template.json? you can consider asking the path from the user

File: automation/vnet/deploy-vnet.ps1 [hunk 5]
```diff
@@ -4,55 +4,70 @@
...
+
+#Hashtable containing parameters for virtual network template
+$VirtualNetworkParameters= @{
+    vnetName = ("nsc-vnet-dev-{0}-{1}-test" -f $RGLocation,$VirtualNetworkName).ToLower();
```
Comment (OyunominJuno) [inline automation/vnet/deploy-vnet.ps1]: hard coded name here, is there a way to ask this from the user?

**Comment by elb-dev** [comment]
Code looks nice and readable, and the script works! It's a lot easier to follow what to do this time around, as opposed to last time. I have just one concern: I think it would make more sense for something like the two names to be set to not required, so they're easy to deploy with default parameters as required.

**Review by elb-dev — APPROVED** [review]
Ohp, already left a comment. Check what I said previously. I think it is good though.

File: automation/vnet/template.json [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      }
+    },
+    "subnet1Name": {
+      "type": "string",
+      "metadata": {
+        "description": "Subnet 1 Name"
+      }
+    },
+    "location": {
+      "type": "string",
```
Comment (asiagromasia) [inline automation/vnet/template.json:32]: adding here allowed values option would limit user input and make it more safe

**Review by OyunominJuno — APPROVED** [review]
this script successfully creates a virtual network and subnet within a resource group.

File: automation/vnet/template.json (hunk 6 above)
Comment (sunshinejen) [reply automation/vnet/template.json:32]: Great idea! I have added your requested changes.

File: automation/vnet/deploy-vnet.ps1 (hunk 1 above)
Comment (sunshinejen) [reply automation/vnet/deploy-vnet.ps1:7]: Good catch, it is updated thanks Juno

**Review by asiagromasia — APPROVED** [review]
easy to read and deploys vnet with subnet

**Pull request merged**